src/usps_processor/api.clj

```diff
@@ -51,26 +52,52 @@
         mailings (d/match-entities db mailing-constraints)]
     mailings))
 
-(defn on-single-match [matches match->body]
-  (case (count matches)
-      1 (-> matches
-            first
-            match->body
-            edn-response)
-      0 (-> "Not found"
-          edn-response
-          (assoc :status 404))
-      (-> "Multiple matches found"
-          edn-response
-          (assoc :status 422))))
+(def standard-not-found
+  (-> "Not found"
+      edn-response
+      (assoc :status 404)))
+
+(def standard-multiple-matches
+  (-> "Multiple matches found"
+      edn-response
+      (assoc :status 422)))
+
+(defn get-scanned-since [req]
+  (when-let [string (get-in req [:params :scanned-since])]
+    (instant/read-instant-date string)))
 
 (defn latest-scan [req]
-  (on-single-match (lookup-mailings req)
-                   (comp render-scan mailing/latest-scan)))
+  (let [matches (lookup-mailings req)]
```

Comment:
I think you want to apply the time filtering to matches here before going into the conditional. It should be ok for there to be multiple matches in general if all but one are before the time cut-off, for example.